doc/stdlib/linalg.rst

```diff
@@ -985,42 +1044,108 @@ Linear algebra functions in Julia are largely implemented by calling functions f
 
    Lower triangle of a matrix.
 
-.. function:: tril(M, k)
+   .. doctest::
+
+       julia> a = ones(4,4)
+       4×4 Array{Float64,2}:
+        1.0  1.0  1.0  1.0
+        1.0  1.0  1.0  1.0
+        1.0  1.0  1.0  1.0
+        1.0  1.0  1.0  1.0
+
+       julia> tril(a)
+       4×4 Array{Float64,2}:
+        1.0  0.0  0.0  0.0
+        1.0  1.0  0.0  0.0
+        1.0  1.0  1.0  0.0
+        1.0  1.0  1.0  1.0
+
+.. function:: tril(M, k::Integer)
 
    .. Docstring generated from Julia source
 
    Returns the lower triangle of ``M`` starting from the ``k``\ th superdiagonal.
 
+   .. doctest::
+
+       julia> a = ones(4,4)
+       4×4 Array{Float64,2}:
+        1.0  1.0  1.0  1.0
+        1.0  1.0  1.0  1.0
+        1.0  1.0  1.0  1.0
+        1.0  1.0  1.0  1.0
+
+       julia> tril(a,3)
+       4×4 Array{Float64,2}:
+        1.0  1.0  1.0  1.0
+        1.0  1.0  1.0  1.0
+        1.0  1.0  1.0  1.0
+        1.0  1.0  1.0  1.0
+
+       julia> tril(a,-3)
+       4×4 Array{Float64,2}:
+        0.0  0.0  0.0  0.0
+        0.0  0.0  0.0  0.0
+        0.0  0.0  0.0  0.0
+        1.0  0.0  0.0  0.0
+
 .. function:: tril!(M)
 
    .. Docstring generated from Julia source
 
-   Lower triangle of a matrix, overwriting ``M`` in the process.
+   Lower triangle of a matrix, overwriting ``M`` in the process. See also :func:`tril`\ .
 
-.. function:: tril!(M, k)
+.. function:: tril!(M, k::Integer)
 
    .. Docstring generated from Julia source
 
    Returns the lower triangle of ``M`` starting from the ``k``\ th superdiagonal, overwriting ``M`` in the process.
 
-.. function:: diagind(M[, k])
+.. function:: diagind(M, k::Integer=0)
 
    .. Docstring generated from Julia source
 
-   A ``Range`` giving the indices of the ``k``\ th diagonal of the matrix ``M``\ .
+   A :class:`Range` giving the indices of the ``k``\ th diagonal of the matrix ``M``\ .
+
+   .. doctest::
+
+       julia> A = [1 2 3; 4 5 6; 7 8 9]
+       3×3 Array{Int64,2}:
+        1  2  3
+        4  5  6
+        7  8  9
+
+       julia> diagind(A,-1)
+       2:4:6
 
-.. function:: diag(M[, k])
+.. function:: diag(M, k::Integer=0)
 
    .. Docstring generated from Julia source
 
-   The ``k``\ th diagonal of a matrix, as a vector. Use ``diagm`` to construct a diagonal matrix.
+   The ``k``\ th diagonal of a matrix, as a vector. Use :func:`diagm` to construct a diagonal matrix.
 
-.. function:: diagm(v[, k])
+   .. doctest::
+
+       julia> diag(A,1)
```

Comment:
Yes but I shouldn't rely on that happening. If you read this in isolation it'd be confusing.